### PR TITLE
Testing section

### PR DESCRIPTION
**Pull request by colin-ch**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -42,6 +42,13 @@ That way you are in control of the order in which things are executed.
 
 
 ## Testing
+There should be unit tests for your code. A unit test should be independent and do an assert for each unit. The test is independent so it has no dependencies on the database or external services. An exception is a repository test -- there you want to test a database transaction.
```
Comment (thvd) [inline readme.md]: "The test is independent so it has no dependencies on the database or external services"
Shouldn't that be: "The tests for business logic should be independent so it has no dependencies on database or external services"

Comment (thvd) [inline readme.md]: @colin-ch can you also write something about "what is the purpose of unit tests"?

Comment (thvd) [reply readme.md]: Ah i see there is already something for that

File: readme.md [hunk 2]
```diff
@@ -42,7 +42,18 @@ That way you are in control of the order in which things are executed.
...
 
+The unit test should be clean and treated as part of the codebase. They are part of your deployment process.
+
+The purpose of a unit test -- it should give quick feedback to check if the production code does what we expect it to do. The unit test is not responsible for testing the infrastructure and application chain. We don’t have to test the framework. The unit test does not know in what context the code will run, so therefore it should be independent and mocking should always be used when necessary.
+
+For example, when testing functionality that is called from a trigger, you want to unit-test the scoped pieces of code that will be executed separately. For example, a unit test will not test the whole process of a record going through a trigger, or a process from UI to database; a unit test is not the place to do that.
+
+A unit test will also help with designing your code. With TDD you will write unit tests and scenarios. After that, you will improve the code, so that it will make only these tests pass.
```
Comment (thvd) [inline readme.md]: "so that it will make only these tests pass"? what do you mean with that?

Comment (colin-ch) [reply readme.md]: You write code only with the purpose to make the tests green with TDD. So no unnecessary code.

File: readme.md [hunk 3]
```diff
@@ -42,15 +42,15 @@ That way you are in control of the order in which things are executed.
 
 
 ## Testing
-There should be unit tests for your code. A unit test should be independent and do an assert for each unit. The tests for business logic should be independent so it has no dependencies on database or external services. An exception is a repository test -- there you want to test a database transaction.
+There should be unit tests for your code. A unit test should be independent and do an assert for each unit. The tests for business logic should be independent so it has no dependencies on databases or external services. An exception is a repository test -- there you want to test a database transaction.
 
-The unit test should be clean and treated as part of the codebase. They are part of your deployment process.
+The unit tests should be clean and treated as part of the codebase. They are part of your deployment process.
```
Comment (thvd) [inline readme.md]: Without "The"?

File: readme.md [hunk 4]
```diff
@@ -42,7 +42,18 @@ That way you are in control of the order in which things are executed.
 
 
 ## Testing
+There should be unit tests for your code. A unit test should be independent and do an assert for each unit. The tests for business logic should be independent so it has no dependencies on databases or external services. An exception is a repository test -- there you want to test a database transaction.
```
Comment (sanderott) [inline readme.md:45]: Do we have a set structure that we follow for our unit tests?

Comment (colin-ch) [reply readme.md:45]: Arrange, Act, Assert

**Review by thvd — APPROVED** [review]

**Pull request merged**